src/sage_interface_internal.h

```diff
@@ -0,0 +1,45 @@
+/* LibGAP - a shared library version of the GAP kernel
+ * Copyright (C) 2013 Volker Braun <vbraun.name@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA. 
+ */
+
+
+#ifndef LIBGAP_INTERNAL__H
+#define LIBGAP_INTERNAL__H
+
+/* Allow environment access to OSX dylib, see http://trac.sagemath.org/14038 */
+#ifdef __APPLE__
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#else
+extern char** environ;
+#endif /* __APPLE__ */
```

Comment:
Again, using `__APPLE__` is not so great, nor that #define. Of course we can leave it for now. But I'd add a FIXME comment, and at some point, we should review whether we remove or at least isolate our use of the global variable `environ` properly, into a single C file (and then this hack should move there).